modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java

```diff
@@ -663,28 +667,39 @@ public ComponentModel getRootComponentModel() {
    * Goes through the entire xml mule application looking for the message processors that can be expanded, and then takes
    * care of the global elements.
    *
-   * @param moduleExtensionStore extensions that will use to check if the element has to be expanded.
+   * @param extensionManager extensions that will be used to check if the element has to be expanded.
    */
-  private void expandModules(Optional<ModuleExtensionStore> moduleExtensionStore) {
-    if (moduleExtensionStore.isPresent()) {
-      createOperationRefEffectiveModel(moduleExtensionStore.get());
-      createConfigRefEffectiveModel(moduleExtensionStore.get());
+  private void expandModules(Optional<ExtensionManager> extensionManager) {
```

Comment:
Is it possible to move all this logic related to modules to another class?. To at least encapsulate all XML modules related logic. At some point we will do a big refactor over ApplicationModel as a whole.